nav2_mppi_controller/src/critics/path_angle_critic.cpp

```diff
@@ -58,17 +58,26 @@ void PathAngleCritic::score(CriticData & data)
   const float goal_x = xt::view(data.path.x, offseted_idx);
   const float goal_y = xt::view(data.path.y, offseted_idx);
 
-  if (utils::posePointAngle(data.state.pose.pose, goal_x, goal_y) < max_angle_to_furthest_) {
+  double forward_angle = utils::posePointAngle(data.state.pose.pose, goal_x, goal_y);
+  double backward_angle = abs(angles::normalize_angle(forward_angle + M_PI));
```

Comment:
I think we need to make a parameter if multidirectionality is valid. If we have preferences for forward, we should not be skipping based on the path going in the opposite direction of the current orientation. We need to apply it in that case (that's actually the most extreme case of why this critic exists to help push that orientation motion).

Reply:
See new commit, by using the plan poses orientation info, we can support all use cases transparently:
- following in the forward direction a plan which has poses with orientations pointing in the direction of path increment
- following in the backward direction a plan which has poses with orientations pointing in the opposite direction of path increment
- doing an inversion manoeuvre  when starting to follow in the backward direction a plan which has poses with orientations pointing in the direction of path increment
- doing an inversion manoeuvre  when starting to follow in the forward direction a plan which has poses with orientations pointing in the opposite direction of path increment

Reply:
Not all planners provide the orientation though - like [NavFn](https://github.com/ros-planning/navigation2/blob/main/nav2_navfn_planner/src/navfn_planner.cpp#L426-L429) or [Theta*](https://github.com/ros-planning/navigation2/blob/main/nav2_theta_star_planner/src/theta_star_planner.cpp#L214-L216) just set the XY coordinates, not the full SE2 poses (because its impossible for them to - they know nothing of kinematics). 

It is a slick solution though if we had that :/ I suppose we could update those 2 planners to use orientations with something like https://github.com/ros-planning/navigation2/blob/main/nav2_smac_planner/src/smoother.cpp#L284-L300 (broken out into a `nav2_util`)